python/modules/objects.py

```diff
@@ -0,0 +1,201 @@
+# coding: utf8
+
+from modules.utils import pos_to_vertices
+import matplotlib.pyplot as plt
+
+print(">> Loading module: 'objects'...")
+
+
+class Object3d:
+
+    def __init__(self, name, dimensions):
+        """
+        Object3d class constructor.
+        """
+        self._name = name
+        self._position = Coord6d([0,0,0,0,0,0])
+        self._dimensions = dimensions
+        self.vertices = pos_to_vertices(self._position, self._dimensions)
+
+    def __str__(self):
+        """
+        Prettily display an Object3d.
+        """
+        str_dim = "| > Dimensions: {}\n".format(self.dimensions)
+        str_pos = "| > Position: {}\n|             {}\n".format(
+                  self.position.spatial, self.position.angular)
+        str_vertices = "| > Vertices:\n"
+        for vertex in self.vertices:
+            str_vertices += '|   '
+            str_vertices += str(vertex) + '\n'
+        return "\n| {}:\n{}{}{}".format(self.name, str_dim, str_pos, str_vertices)
+
+    @property
+    def position(self):
+        return self._position
+
+    @property
+    def dimensions(self):
+        return self._dimensions
+
+    @property
+    def name(self):
+        return self._name
+
+
+class Mobile(Object3d):
+
+    def __init__(self, name, dimensions):
+        """
+        Mobile class constructor.
+        """
+        Objet3d.__init__(self, name, dimensions)
+
+    def move(self, destination):
+        """
+        Move a mobile.
+        """
+        # change position
+        self.position.move(destination)
+        # refresh the vertices
+        self.vertices = pos_to_vertices(self.position, self.dimensions)
+
+
+class Hangar(Object3d):
+
+    def __init__(self, name, dimensions):
+        """
+        Hangar class constructor.
+        """
+        Objet3d.__init__(self, name, dimensions)
+
+
+class Coord3d:
+
+    def __init__(self, coord=[0,0,0]):
+        """
+        Coord3d class constructor.
+        """
+        x, y, z = coord
+        self.x = x
+        self.y = y
+        self.z = z
+        self.spatial = [x, y, z]
+
+    def __str__(self):
+        """
+        Prettily display a Coord3d object.
+        """
+        return "({0:.3f}, {1:.3f}, {2:.3f})".format(self.x, self.y, self.z)
+
+    def move(self, destination):
+        """
+        Change the coordinates.
+        """
+        self.__init__(destination)
```

Comment:
We have to choose whether working with arrays [x,y,z,a,b,g] as a single attribute or with 6 coordinate attributes, and I cannot decide myself yet, I think you can feel it while reading the code.
I used self.__init__(destination) because with decorators, I would have written 6 decorators, one for each coordinate.
But I agree, it's not very clean!
What do you think about it?

Reply:
I think that what you did with Coord6d is the right thing : 
 - When working with objects, you modify the 6d coordinates with the dedicated methods as if it were one block;
 - When performing more complex calculations, you take the numpy array from Coord6d and modify that array directly (coordinate by coordinate if you need to);